src/task.ts

```diff
@@ -91,41 +84,51 @@ export default class Task extends EventTarget {
         }
 
         samples.push(taskTime);
-        this.runs += 1;
         totalTime += taskTime;
 
         if (this.opts.afterEach != null) {
           await this.opts.afterEach.call(this);
         }
       }
-    } catch (e) {
-      this.setResult({ error: e });
-      if (this.bench.throws) {
-        throw e;
-      }
+    } catch (error) {
+      return { error };
     }
 
     if (this.opts.afterAll != null) {
       try {
         await this.opts.afterAll.call(this);
-      } catch (e) {
-        this.setResult({ error: e });
+      } catch (error) {
+        return { error };
```

Comment:
it is possible that incorrect completion of the task in afterAll will cause incorrect initialization in the next task
in which case the result cannot be trusted

That is why it was decided to return an error instead of a value